operators/src/pro/cache/cache_chunks.rs

```diff
@@ -391,31 +320,337 @@ impl_cache_result_check!(MultiPolygon);
 
 impl<G> CacheElement for FeatureCollection<G>
 where
-    G: Geometry + ArrowTyped,
-    FeatureCollection<G>: ByteSize
-        + CacheElementHitCheck
-        + CacheBackendElementExt<
-            Query = VectorQueryRectangle,
-            LandingZoneContainer = LandingZoneQueryFeatures,
-            CacheContainer = CachedFeatures,
-        >,
+    G: Geometry + ArrowTyped + 'static,
+    CompressedFeatureCollection<G>: CacheBackendElementExt<
+        Query = VectorQueryRectangle,
+        LandingZoneContainer = LandingZoneQueryFeatures,
+        CacheContainer = CachedFeatures,
+    >,
+    FeatureCollection<G>: ByteSize + CacheElementSpatialBounds,
 {
-    type StoredCacheElement = FeatureCollection<G>;
+    type StoredCacheElement = CompressedFeatureCollection<G>;
     type Query = VectorQueryRectangle;
-    type ResultStream = CacheChunkStream<G>;
+    type ResultStream =
+        CacheStream<CompressedFeatureCollection<G>, FeatureCollection<G>, VectorQueryRectangle>;
 
     fn into_stored_element(self) -> Self::StoredCacheElement {
-        self
+        CompressedFeatureCollection::<G>::from_collection(self)
+            .expect("Compressing the feature collection should not fail")
     }
 
     fn from_stored_element_ref(stored: &Self::StoredCacheElement) -> Result<Self, CacheError> {
-        Ok(stored.clone())
+        stored.to_collection()
     }
 
     fn result_stream(
         stored_data: Arc<Vec<Self::StoredCacheElement>>,
         query: Self::Query,
     ) -> Self::ResultStream {
-        CacheChunkStream::new(stored_data, query)
+        CacheStream::new(stored_data, query)
+    }
+}
+
+pub type CompressedFeatureCollection<G> = CompressedFeatureCollectionImpl<G, Lz4FlexCompression>;
+pub type CompressedDataCollection = CompressedFeatureCollection<NoGeometry>;
+pub type CompressedMultiPointCollection = CompressedFeatureCollection<MultiPoint>;
+pub type CompressedMultiLineStringCollection = CompressedFeatureCollection<MultiLineString>;
+pub type CompressedMultiPolygonCollection = CompressedFeatureCollection<MultiPolygon>;
+
+#[derive(Debug)]
+pub struct CompressedFeatureCollectionImpl<G, C> {
+    data: Vec<u8>,
+    spatial_bounds: Option<geoengine_datatypes::primitives::BoundingBox2D>,
+    time_interval: Option<geoengine_datatypes::primitives::TimeInterval>,
```

Comment:
why is this an option? Features always have an attached time. It can just be [begin of time, end of time). Does this require special handling?

Reply:
The option handles the case when the collection is empty